File: applications/ios/DiceRoller/DiceRollRepository/RollResponse.swift
struct RollResponse {
    let diceValues: [UInt32]

    func toProtobuf() -> ProtobufRoll_ProtobufRollResponse {
        let protobufRollResponse = ProtobufRoll_ProtobufRollResponse.with {
            $0.diceValues = diceValues
        }

        return protobufRollResponse
    }

    static func fromProtobuf(protobufRollResponse: ProtobufRoll_ProtobufRollResponse) -> RollResponse {
        return RollResponse(diceValues: protobufRollResponse.diceValues)
    }
}

File: platform-wrappers/android/LibDiceRollerAdapter/src/main/kotlin/com/walterscarborough/libdiceroller/RollResponse.kt
package com.walterscarborough.libdiceroller

import protobuf_roll.RollResponse.ProtobufRollResponse

data class RollResponse(val diceValues: List<UInt>) {
    fun toProtobuf(): ProtobufRollResponse {
        return ProtobufRollResponse
            .newBuilder()
            .addAllDiceValues(this.diceValues.map(UInt::toInt))
            .build()
    }

    companion object {
        fun fromProtobuf(protobufRollResponse: ProtobufRollResponse): RollResponse {
            return RollResponse(
                diceValues = protobufRollResponse.diceValuesList.map(Int::toUInt)
            )
        }
    }
}
